packages/components/src/components/DetailsPanel.vue

```diff
@@ -109,6 +118,14 @@ export default {
         this.filtersRootObjects.includes(this.selectedObject.fqid)
       );
     },
+    sourceLocationObject() {
+      if (this.appMap.metadata.source_location) {
+        return {
+          location: this.appMap.metadata.source_location,
+        };
+      }
+      return null;
+    },
```

Comment:
The source location cannot be static because we need the ability to modify the location based off the environment. For example, in an IDE the location may be a local file whereas the browser would require a link to the file in a remote repository, e.g. GitHub, GitLab, etc. Providing this as a static value does not allow us to specify this behavior.

Reply:
@dustinbyrne Link to CodeObject location is always a link to file which will be opened in IDE, are we need another behavior for AppMap source location?
![image](https://user-images.githubusercontent.com/1647695/165349720-ee38a0d3-29ad-46cc-81ef-9d6d4a39a021.png)

Reply:
Only in some cases will the link need to be opened in an IDE. In other cases, we'd need to transform the file path to a location in GitHub. 

See below for an example of how we're using this event protocol to resolve locations in the AppMap data into something else like a URL:
https://github.com/applandinc/appmap-js/blob/main/packages/components/src/stories/support/resolvePath.js

Reply:
Thanks for explanation, it's done now. https://github.com/applandinc/appmap-js/pull/569/commits/7f462f1d4a4ba54450f3619c8d466d1ab55d0b12